solutions/CF Mixing Water.mdx

```diff
@@ -0,0 +1,118 @@
+---
+id: cf-1359C
+source: CF
+title: Mixing Water
+author: Jesse Choe
+---
+
+<Info title="Constant Time Solution">
+**Note: the $\mathcal{O}(1)$ solution for this problem is provided in the
+[official editorial](https://codeforces.com/blog/entry/78116).**
+</Info>
+
+Consider the following observations:
+
+- If the number of cups of hot water is equal to the number of cups of cold water in the barrel, then the minimum number of cups required to pour into the barrel is exactly $2$.
+- The number of cups of hot water must be greater than the number of cups of cold water if the previous condition isn't satisfied. Thus, there will be $c + 1$ cups of hot water and $c$ cups of cold water.
+- The two barrels with the closest temperature to $t$ will be the barrels with temperatures slightly more and slightly less than $t$. The two barrels will be $a$ and $b$, respectively.
```

Comment:
Can delete this and say smth like since temperature func of c is decreasing we can binsearch to approximate it.

Reply:
ig you'll have to change the stuff below to flow better

Reply:
> Can delete this and say smth like since temperature func of c is decreasing we can binsearch to approximate it.

What?

Reply:
the temperature is l+(h-l)(c+1)/(2c+1) when c is odd. THis is a decreasing function (aka monotonic); since it's monotonic we can binsearch to approximate the best value for $c$, and test the odd integers aroudn it